frontend/src/pages/Question.tsx

```diff
@@ -1,11 +1,65 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import SubmissionWidget from "../components/SubmissionWidget";
+import { QuestionsPublic, Question, Document } from "../models/questions";
+import LeftSideBar from "../components/LeftSideBar";
+import Markdown from "react-markdown";
+
+const QuestionPage = () => {
+  const [questions, setQuestions] = useState<Question[] | null>(null);
+  const [selectedQuestion, setSelectedQuestion] = useState<Question | null>(
+    null
+  );
+  const [globalDocs, setGlobalDocs] = useState<Document[]>([]);
+
+  useEffect(() => {
+    fetch("/api/questions", {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${localStorage.getItem("token")}`,
+      },
+    })
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(`HTTP Error with Status Code: ${response.status}`);
+        }
+        return response.json();
+      })
+      .then((responseData: QuestionsPublic) => {
+        setQuestions(responseData.questions);
+        setGlobalDocs(responseData.global_docs);
+        setSelectedQuestion(responseData.questions[0]);
+      })
+      .catch((error) => {
+        console.error("Error:", error);
+      });
+  }, []);
+
+  // State stores the selected question.
+  const handleQuestionClick = (questionNum: number) => {
+    const question = questions?.find((q) => q.num - 1 === questionNum) || null;
+    setSelectedQuestion(question);
```

Comment:
You _might_ be able to take advantage of the react routers sub routes to do this as well. I'm unsure if it'll work though so just a suggestion.

Reply:
I think this was Matthew's change. I'll leave this open for him. I had a different way of doing it.